src/main/scala/viper/gobra/frontend/Desugar.scala

```diff
@@ -2263,6 +2263,24 @@ object Desugar {
               } yield in.UneqCmp(l, r)(src)
             }
 
+          case PGhostEquals(left, right) =>
+            if (info.typ(left) == PermissionT || info.typ(right) == PermissionT) {
+              // TODO: inherits the same check as for PEquals and PUnequals. Could possibly be dropped or
+              //       refactored in the future
+              for { l <- permissionD(ctx, info)(left); r <- permissionD(ctx, info)(right) } yield in.GhostEqCmp(l, r)(src)
+            } else {
+              for { l <- exprD(ctx, info)(left); r <- exprD(ctx, info)(right) } yield in.GhostEqCmp(l, r)(src)
+            }
```

Comment:
I do not understand why the extra permissions cse is necessary since they should also just be expressions. Does something break, when you remove the case distinction?

Reply:
This guarantees that, when you are comparing a perm expression to some literal, both operands are treated as permissions. Otherwise, encoding the following comparison would fail if the right hand side is treated as an int (integer division is used here):
```
func test() {
    assert perm(1/2) == 1/2
}
```
Another technical point is that this expression would be encoded as a comparison between an integer and a perm, for which we do not implement a case (although we could).

Reply:
Please add a comment explaining the first part "This guarantees that, when you are comparing a perm expression to some literal, both operands are treated as permissions. E.g. the right-hand side of perm(1/2) == 1/2 is treated as a permission.

To me this seems like bad language design on our part, but we can discuss this in a future meeting and decide on a solution then.